### ISSUE TITLE
describe_optimized_plan

### ISSUE THREAD

**Issue by braaannigan** [pola-rs/polars#3900]
Having `describe_plan` and `describe_optimized_plan` is confusing for newcomers and verbose. Propose to deprecate these and replace them with something like:
```python
def plan(
        optimized:bool = True,
        type_coercion: bool = True,
        predicate_pushdown: bool = True,
        projection_pushdown: bool = True,
        simplify_expression: bool = True,
        slice_pushdown: bool = True
)
```

**Comment by alexander-beedie** [comment]
I'd keep the name `describe_plan` (as it does indeed return a description of the plan), but combine the params into the one function, as suggested. (Just `plan` is a bit too ambiguous).

**Comment by braaannigan** [comment]
> If combining the two functions I'd keep the name `describe_plan` (as it does indeed return a description of the plan, and it's what people are used to), adding the params as suggested (though don't need 'optimized'). Just `plan` is a bit ambiguous.

Does it not need the optimized param to allow the non-optimized plan to be returned?

**Comment by alexander-beedie** [comment]
> Does it not need the optimized param to allow the non-optimized plan to be returned?

No; I believe it would just need all the other params set to `False`, as they _are_ the optimizations.

**Comment by braaannigan** [comment]
> > Having describe_plan and describe_optimized_plan is confusing
I'm giving a workshop this week and find I'm having to say "ok, ignore this describe_plan if you want to know what's going to happen because it's not actually the plan"

**Comment by ritchie46** [comment]
> I'm giving a workshop this week and find I'm having to say "ok, ignore this describe_plan if you want to know what's going to happen because it's not actually the plan"

It does have merit. It is the query plan 'as is'. Comparing the two shows which optimizations polars does. We often have to compare them both.

**Comment by stinodego** [comment]
This method is now called `explain` and it has a parameter `optimized=True/False` defaulting to `True`.

I believe this issue has been adequately addressed. If not, please open a new issue.
